src/sql/workbench/contrib/notebook/browser/models/notebookModel.ts

```diff
@@ -457,14 +451,8 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			}
 			let profile = new ConnectionProfile(this._notebookOptions.capabilitiesService, this.connectionProfile);
 
-			// TODO: this code needs to be fixed since it is called before the this._savedKernelInfo is set.
-			// This means it always fails, and we end up using the default connection instead. If you right-click
```

Comment:
>fails [](start = 27, length = 5)

So is this not a problem anymore then? What happens if you right click on a disconnected server and do new notebook now?

Reply:
Correct. I tested this scenario using the following test cases:
- No servers are connected from OE, right click on various servers and choose New Notebook
- One server is connected from OE, right click on both the connected and disconnected servers. Run a query to ensure that the right server is being targeted.